### PR TITLE
Time widget misc2

### PR DESCRIPTION
**Pull request by WenzDaniel**
**What is the problem / what does the code in this PR do**
In this PR we use APs ipython time and date widget example and convert it into a class. I extended the class a bit such that the timestamps are timezone aware. The user can select between CET and UTC.

**Can you give a minimal working example (or illustrate with a figure)?**
```
tw = TimeWidgets()
tw.create_widgets()
start, end = tw.get_start_end()
```

### CODE REVIEW

File: straxen/misc.py [hunk 1]
```diff
@@ -79,3 +88,103 @@ def utilix_is_configured(header='RunDB', section='pymongo_database') -> bool:
                 straxen.uconfig.get(header, section) is not None)
     except NoSectionError:
         return False
+
+@export
```
[the remark this answers is not in the document]
Comment (jorana) [reply straxen/misc.py:92]: 👍

File: straxen/misc.py [hunk 2]
```diff
@@ -79,3 +88,103 @@ def utilix_is_configured(header='RunDB', section='pymongo_database') -> bool:
...
+class TimeWidgets:
```
[the remark this answers is not in the document]
Comment (jorana) [reply straxen/misc.py:93]: 👍

[the remark this answers is not in the document]
Comment (jorana) [reply straxen/misc.py:93]: 👎

File: straxen/misc.py [hunk 3]
```diff
@@ -79,3 +88,103 @@ def utilix_is_configured(header='RunDB', section='pymongo_database') -> bool:
...
+        end = end.timestamp()
+        end = int(end * 10**9)
+
+        if start > end:
+            warnings.warn('Start time is larger than endtime are you '
+                          'sure you wanted this?')
+        return start, end
+
+    @staticmethod
+    def _create_date_and_time_widget(date_and_time, widget_describtion):
```
[the remark this answers is not in the document]
Comment (jorana) [reply straxen/misc.py:148]: 👎

File: straxen/misc.py [hunk 4]
```diff
@@ -79,3 +88,103 @@ def utilix_is_configured(header='RunDB', section='pymongo_database') -> bool:
...
+        date = datetime(date_and_time.year,
+                        date_and_time.month,
+                        date_and_time.day)
+        date = widgets.DatePicker(description=widget_describtion,
+                                  value=date,
+                                  layout=Layout(width='225px'),
+                                  disabled=False)
+
+        time = "{:02d}:{:02d}".format(int(date_and_time.hour),
```
[the remark this answers is not in the document]
Comment (jorana) [reply straxen/misc.py:157]: 👎

File: straxen/misc.py [hunk 5]
```diff
@@ -79,3 +88,103 @@ def utilix_is_configured(header='RunDB', section='pymongo_database') -> bool:
...
+                                      int(date_and_time.minute))
+        time = widgets.Text(value=time,
+                            layout=Layout(width='75px'),
+                            disabled=False)
+
+        return widgets.HBox([date, time])
+
+    @staticmethod
+    def _create_time_zone_widget():
+        _time_zone_widget = widgets.Dropdown(options=[('CET', 0), ('UTC', 1)],
```
[the remark this answers is not in the document]
Comment (jorana) [reply straxen/misc.py:167]: 👍

File: straxen/misc.py [hunk 6]
```diff
@@ -79,3 +88,103 @@ def utilix_is_configured(header='RunDB', section='pymongo_database') -> bool:
...
+                                             value=0,
+                                             description='Time Zone:',
+                                             )
+        return _time_zone_widget
+
+    @staticmethod
+    def _convert_to_datetime(time_widget, time_zone):
+        """
+        Converts
```
Comment (jorana) [inline straxen/misc.py:176]: Missing?

File: straxen/misc.py [hunk 7]
```diff
@@ -79,3 +88,103 @@ def utilix_is_configured(header='RunDB', section='pymongo_database') -> bool:
...
+        self._start_widget = self._create_date_and_time_widget(utcstart, 'Start')
+        self._end_widget = self._create_date_and_time_widget(utcend, 'End')
+        self._time_zone_widget = self._create_time_zone_widget()
+        self._created_widgets = True
+
+        return widgets.VBox([self._time_zone_widget,
+                             self._start_widget,
+                             self._end_widget])
+
+    def get_start_end(self):
```
Comment (jorana) [inline straxen/misc.py:123]: Did you not have a similar function in the scada interface, if so, can we reuse?

**Review by jorana — COMMENTED** [review]
Thanks @WenzDaniel , looks simple enough. What do you forsee as the use case?

I can imagine datetime.datetime(2021,04,01,0,0,0).timestamp()*1e9 works just as well and allows you better precision for the timestamps.

**Comment by WenzDaniel** [comment]
The use case would be mostly be slow control related. But in principle it can be handy for every application in which someone monitors a rate on a human readable time axes who would like to check things in data afterwards. 
At least it happened to me more than once that I had to search for a unix time converter.

File: straxen/misc.py (hunk 7 above)
Comment (WenzDaniel) [reply straxen/misc.py:123]: In scada there is a function to convert the time index of a pandas dataframe into different time_zones. So unfortunately not.

**Pull request merged**